src/Core/Services/DetermineStatusCodeMiddleware.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Net;
+using Azure.DataApiBuilder.Service.Exceptions;
+using HotChocolate.Execution;
+
+/// <summary>
+/// The VerifyResultMiddleware inspects the IExecutionResult created by HotChocolate
+/// and determines the appropriate HTTP error code to return based on the errors in the result.
+/// By Default, without this serializer, HotChocolate will return a 500 status code when database errors
+/// exist. However, there is a specific error code we check for that should return a 400 status code:
+/// - DatabaseInputError. This indicates that the client can make a change to request contents to influence
+/// a change in the response.
+/// </summary>
+public sealed class DetermineStatusCodeMiddleware(RequestDelegate next)
+{
+    private const string ERROR_CODE = nameof(DataApiBuilderException.SubStatusCodes.DatabaseInputError);
```

Comment:
This is very specific. Do we also http status code for auth errors or conflict errors or other bad request statuses in DataApiBuilderException?

Reply:
This is an status code determination:
https://github.com/Azure/data-api-builder/blob/7f8fa6b4a7468582e6015df96f75f3cf8c0987f9/src/Core/Services/DabGraphQLResultSerializer.cs#L12 
not relevant to the migration done in this PR. 

Auth errors will automatically result in 401 already. We could potentially look at all possible other exception states in DABException and map it to BadRequest - some of them states apply only to REST APIs. This should be tackled in a separate issue.